### PR TITLE
operator [R] [CI] hpe-csi-operator (2.2.1 2.3.0)

### PR DESCRIPTION
**Pull request by sijeesh**
Thanks submitting your Operator. Please check below list before you create your Pull Request.

### New Submissions

* [ ] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-via-pr.md)?
* [ ] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [ ] Have you tested your Operator with all Custom Resource Definitions?
* [ ] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [ ] Have you considered whether you want use [semantic versioning order](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md#semver-mode)?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?
* [ ] Is operator [icon](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#operator-icon) set?

### Updates to existing Operators

* [ ] Did you create a `ci.yaml` file according to the [update instructions](https://github.com/operator-framework/community-operators/blob/master/docs/operator-ci-yaml.md)?
* [ ] Is your new CSV pointing to the previous version with the `replaces` property if you chose `replaces-mode` via the `updateGraph` property in `ci.yaml`?
* [ ] Is your new CSV referenced in the [appropriate channel](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#channels) defined in the `package.yaml` or `annotations.yaml` ?
* [ ] Have you tested an update to your Operator when deployed via OLM?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing-prerequisites.md#sign-your-work)?

### Your submission should not

* [ ] Modify more than one operator
* [ ] Modify an Operator you don't own
* [ ] Rename an operator - please remove and add with a different name instead
* [ ] Modify any files outside the above mentioned folders
* [ ] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [ ] Description about the managed Application and where to find more information
2. [ ] Features and capabilities of your Operator and how to use it
3. [ ] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [ ] Human readable name and 1-liner description about your Operator
* [ ] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-operator.md#categories)<sup>1</sup>
* [ ] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [ ] Links to the maintainer, source code and documentation
* [ ] Example templates for all Custom Resource Definitions intended to be used
* [ ] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file an issue against this repo and explain your need

<sup>2</sup> For more information see [here](https://sdk.operatorframework.io/docs/overview/#operator-capability-level)

### CODE REVIEW

File: operators/hpe-csi-operator/2.3.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+LABEL operators.operatorframework.io.bundle.channels.v1=stable
+LABEL operators.operatorframework.io.bundle.channel.default.v1=stable
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.0.0
+LABEL operators.operatorframework.io.metrics.mediatype.v1=metrics+v1
+LABEL operators.operatorframework.io.metrics.project_layout=helm.sdk.operatorframework.io/v1
+LABEL operators.operatorframework.io.test.config.v1=tests/scorecard/
+LABEL operators.operatorframework.io.test.mediatype.v1=scorecard+v1
+
+#Add these labels
+LABEL com.redhat.openshift.versions="v4.10,v4.12"
```
Comment (datamattsson) [inline operators/hpe-csi-operator/2.3.0/Dockerfile:16]: This should be "v4.10-v4.12"

File: operators/hpe-csi-operator/ci.yaml [hunk 2]
```diff
@@ -6,3 +6,5 @@ updateGraph: replaces-mode
 reviewers:
   - raunakkumar
   - pavansshanbhag
```
Comment (datamattsson) [inline operators/hpe-csi-operator/ci.yaml:8]: Remove.

**Review by datamattsson — CHANGES_REQUESTED** [review]
Shouldn't the csv file match the operator version? I.e operators/hpe-csi-operator/2.3.0/manifests/hpe-csi-operator.v2.3.0.clusterserviceversion.yaml?

File: operators/hpe-csi-operator/ci.yaml (hunk 2 above)
Comment (sijeesh) [reply operators/hpe-csi-operator/ci.yaml]: Done

File: operators/hpe-csi-operator/2.3.0/Dockerfile (hunk 1 above)
Comment (sijeesh) [reply operators/hpe-csi-operator/2.3.0/Dockerfile]: Done

**Comment by mporrato** [comment]
Hi @sijeesh , you are removing all original reviewers. We will need an approval from at least one of them before authorizing the change.
cc: @raunakkumar

**Comment by mporrato** [comment]
@datamattsson looks like the upgrade test is failing: does the upgrade from 2.2.1 to 2.3.0 work in your test environment?

**Comment by datamattsson** [comment]
@mporrato this Operator has never supported upgrading and probably never will. It's upgraded by completely removing and installing it again.

image

**Comment by datamattsson** [comment]
@mporrato please advise next steps.

**Comment by datamattsson** [comment]
@mvalarh @J0zi please review and merge.

**Comment by datamattsson** [comment]
@mporrato and @Allda, is there anything we need to do on the HPE side to get this merged?

**Comment by datamattsson** [comment]
This has now been sitting here for a month without any comment from the reviewers. Is there anything we can do to help from the HPE side to move this forward?

**Comment by madorn** [comment]
@datamattsson Hi Michael - we reached out to some team members on the RH side for assistance here. Please stand by.

**Comment by datamattsson** [comment]
> Looks like 2.2.0 container images being referenced in 2.2.1 bundle?

Yes, there is no v2.2.1 csi-driver image, the patch release was for the CSV itself only I believe. We're trying to merge v2.3.0 FYI.

**Pull request merged**